libgalois/include/katana/PODVector.h

```diff
@@ -173,9 +220,18 @@ class PODVector {
       capacity_ <<= 1;
     }
 
-    _Tp* new_data_ = static_cast<_Tp*>(
-        realloc(reinterpret_cast<void*>(data_), capacity_ * sizeof(_Tp)));
+    if (pinned_) {
+      if (data_ != nullptr) {
+        munlock(data_, old_bytes);
+      }
+    }
+    const size_t new_bytes = capacity_ * sizeof(_Tp);
+    _Tp* new_data_ =
+        static_cast<_Tp*>(realloc(reinterpret_cast<void*>(data_), new_bytes));
     KATANA_LOG_DEBUG_ASSERT(new_data_);
```

Comment:
I think these should be `KATANA_LOG_ASSERT` the reason is that we are already allocating which is slow and catching allocation failures quickly in some form in release builds is important. Since it can happen if the user doesn't have enough RAM in their system.

Reply:
I've tested and such constructors seem to behave exactly the same. Without the no-argument constructor, the code compiles well, including the places that use `PODVector` with default construction (e.g. inside an `std::vector`). However, if I add an explicit no-argument constructor, I get compilation errors: `error: call of overloaded ‘PODVector()’ is ambiguous`. So there should be no problem here with the code as it is.